src/relayserver/RegistrationManager.ts

```diff
@@ -145,6 +160,12 @@ export class RegistrationManager {
     }
   }
 
+  _extractDuePendingEvents (currentBlock: number): EventData[] {
+    const ret = this.delayedEvents.filter(event => event.block <= currentBlock).map(e => e.eventData)
+    this.delayedEvents = [...this.delayedEvents.filter(event => event.block > currentBlock)]
```

Comment:
Why do you need to use `a = [...x]` instead of `a = x` here? 

Also, this method is getter with side effects. I don't like it. "Extract"..?

Reply:
its an atomic operation: extract some, leave the rest. I deliberately don't want to return some, and leave a chance they will be returned again - hence its "extract" and not a "get"

Reply:
You did not answer the first question.

Also, I don't agree with the answer to the second one.

Reply:
I don't trust "filter" to return a "stable" array. since I keep the result in memory, I want to make sure it is indeed an array, hence the "[...]" wrapper.

I don't want to have 2 methods, one is "getDueItems" and the other "removeDueItems", as its an atomic operation.